src/main/java/org/cbioportal/security/config/Saml2SecurityConfig.java

```diff
@@ -1,49 +1,81 @@
 import org.springframework.security.saml2.provider.service.authentication.Saml2AuthenticatedPrincipal;
 import org.springframework.security.saml2.provider.service.authentication.Saml2Authentication;
+import org.springframework.security.saml2.provider.service.registration.RelyingPartyRegistrationRepository;
+import org.springframework.security.saml2.provider.service.web.DefaultRelyingPartyRegistrationResolver;
+import org.springframework.security.saml2.provider.service.web.authentication.logout.OpenSaml4LogoutRequestResolver;
+import org.springframework.security.saml2.provider.service.web.authentication.logout.Saml2RelyingPartyInitiatedLogoutSuccessHandler;
+import org.springframework.security.web.DefaultSecurityFilterChain;
 import org.springframework.security.web.SecurityFilterChain;
 import org.springframework.security.web.authentication.HttpStatusEntryPoint;
+import org.springframework.security.web.authentication.logout.LogoutSuccessHandler;
 import org.springframework.security.web.util.matcher.AntPathRequestMatcher;
 
 import java.util.Collection;
 import java.util.HashSet;
 import java.util.Objects;
 import java.util.Set;
 
+import static org.springframework.security.config.Customizer.withDefaults;
+
 @Configuration
 @EnableWebSecurity
+@ConditionalOnProperty(value = "authenticate", havingValue = "saml")
 public class Saml2SecurityConfig {
+    
+    @Value("${saml.logout.url:/saml/logout}")
+    private String logoutUrl;
+
+    @Autowired(required = false)
+    private RelyingPartyRegistrationRepository relyingPartyRegistrationRepository;
+
+    @Autowired
+    public void configure(AuthenticationManagerBuilder builder) {
+        OpenSaml4AuthenticationProvider saml4AuthenticationProvider = new OpenSaml4AuthenticationProvider();
+        saml4AuthenticationProvider.setResponseAuthenticationConverter(rolesConverter());
+        builder.authenticationProvider(saml4AuthenticationProvider);
+    }
+    
     @Bean
-    @ConditionalOnProperty(value = "authenticate", havingValue = "saml")
     public SecurityFilterChain samlFilterChain(HttpSecurity http) throws Exception {
-        OpenSaml4AuthenticationProvider authenticationProvider = new OpenSaml4AuthenticationProvider();
-        authenticationProvider.setResponseAuthenticationConverter(rolesConverter());
-        
-        return http.authorizeHttpRequests(auth -> 
-                auth.requestMatchers("/api/health", "/login", "/images/**" ).permitAll()
-                    .anyRequest().authenticated())
-            .exceptionHandling(eh -> 
-                eh.defaultAuthenticationEntryPointFor(new HttpStatusEntryPoint(HttpStatus.UNAUTHORIZED), AntPathRequestMatcher.antMatcher("/api/**")))
-            .saml2Login(saml2 -> saml2
-                .authenticationManager(new ProviderManager(authenticationProvider)))
-            .logout(logout -> logout.logoutSuccessUrl("/login?logout_success"))
+        DefaultSecurityFilterChain build = http
```

Comment:
This is the same answer as under here:

> I think that in this form of OAuth2 login (there is one upstream OIDC IDP) the logout should be transmitted to the IDP. AFAIK the logoutSuccessHandler is invoked after local logout by Spring Security. I included a link to more information on this in a comment:
> 
> // See: https://docs.spring.io/spring-security/reference/5.7-SNAPSHOT/servlet/oauth2/login/advanced.html#oauth2login-advanced-oidc-logout
> 
> If I would not do this. The user would not be able to logout. The following would occur:
> 
> User presses logout button
> cBioPortal screen reloads
> This would be the result of:
> 
> Spring performing local logout
> User is directed to /login
> User is redicted to IDP
> IDP still has active session for user and redirects user to cBioPortal (user does not need to enter username/password)
> cBioPortal screen loads
> So, because of the SSO session with the IDP not being terminated and there is a single IDP gives the user the impression that no actual logout was performed.
> 
> This behavior is not applicable when there are multiple OAuth2 IDP, as you have in your examples (Google, Microsoft). In this case the user is kicked back to the IDP selection window when visiting /login.
> 
> I also do not know how to best solve this. I think there is different behavior needed when having a single or when having multiple OAuth2 IDPs.